packages/interceptors-opentelemetry/src/worker/index.ts

```diff
@@ -17,6 +23,10 @@ export interface InterceptorOptions {
  * provided in the Activity input headers.
  */
 export class OpenTelemetryActivityInboundInterceptor implements ActivityInboundCallsInterceptor {
+  static asFactory(options?: InterceptorOptions): ActivityInterceptorsFactory {
+    return (ctx) => ({ inbound: [new OpenTelemetryActivityInboundInterceptor(ctx, options)] });
+  }
```

Comment:
I'd rather not expose this.

Reply:
Oh, really? TBH I feel quite the opposite...

I want to avoid requiring users to know exactly which kinds of interceptors and sinks need to be instantiated for SDK provided optional features. That evolves badly, as one day, they may need only an activity inbound interceptor, but then, at some point in the future, they would also require an activity outbound interceptor. And even though `ConnectionInjectorInterceptor` is not a public feature, I think applying the same pattern internally make sense.

As it currently stand, I think that gives a reasonably clean API.

    const worker = await Worker.create({
      workflowsPath: require.resolve('./workflows'),
      activities,
      taskQueue: 'test-otel',
      interceptors: {
        workflowModules: [require.resolve('./workflows/otel-interceptors')],
        activity: [
          OpenTelemetryActivityInboundInterceptor.asFactory(),
          ConnectionInjectorInterceptor.asFactory(connection),
        ],
      },
      sinks,
    });

Reply:
Now, that's essentially the same concern that led to the introduction of `WorkflowCoverage.augmentWorkerOptions` and friends, which is arguably an even more complete approach. The only thing I don't really like about it is the fact that it wraps around the whole `WorkerOptions`, which is kinda the objective of course, but has the unwanted side effect of creating some "distance" between the `Worker.create` and the actual options. But I can live with that.

Using that pattern instead of the `asFactory()` one I introduced previously, we could get something like this:

```
    const worker = await Worker.create(
      OpenTelemetryContextPropagator.augmentWorkerOptions(
        ConnectionInjector.augmentWorkerOptions({
          workflowsPath: require.resolve('./workflows'),
          activities,
          taskQueue: 'test-otel',
        })
      )
    );
```

Or, what I consider to be the best of all solutions, something similar to this:

```
    const worker = await Worker.create({
      workflowsPath: require.resolve('./workflows'),
      activities,
      taskQueue: 'test-otel',
      extensions: [
        OpenTelemetryContextPropagator,
        ConnectionInjector,
      ]
    })
```

Reply:
All of that to say that I'm open to reconsider this in the interest of improving DX. In the immediate, though, I'll move on with what I already proposed in this PR. We can revisit that next week.

Reply:
Reverted `asFactory` for the moment. After thinking about this more, I really want to come up an approach that is not limited to activity interceptors, more in the spirit of the "extension" concept I mentioned. This will be done at a later point. In the mean time, it wouldn't be desirable to introduce a new approach (ie. `asFactory`) that would then be deprecated soon after.